packaging/setup/plugins/ovirt-engine-setup/ovirt-engine/network/ovirtproviderovn.py

```diff
@@ -940,10 +940,19 @@ def _print_commands(self, message, commands):
         after=(
             oengcommcons.Stages.ADMIN_PASSWORD_SET,
         ),
-        condition=lambda self: self._enabled,
     )
     def _customization_credentials(self):
-        self._user, self._password = self._get_provider_credentials()
+        if self._enabled:
+            self._user, self._password = self._get_provider_credentials()
+            return
+
+        ovnprovider_installed = self._is_provider_installed()
+        keycloak_enabled = self.environment.get(
+            oengcommcons.KeycloakEnv.KEYCLOAK_ENABLED
+        )
+
+        if ovnprovider_installed and keycloak_enabled:
```

Comment:
With no further condition? Consider this:

1. Run engine-setup without keycloak
2. Run engine-setup enabling keycloak the first time - this is the run for which you are writing the current patch
3. Run again engine-setup, e.g. for an upgrade

It seems like you'll ask for credentials also on (3.).

Reply:
It is only asking for confirmation whether to use the keycloak sso credentials:
`Use default credentials (admin@ovirt@internalsso) for ovirt-provider-ovn (Yes, No) [Yes]:`
After some consideration, I agree it does not make any sense because, in order for ovn provider work correctly, it needs the keycloak sso credentials. I will make an update to include this requirement and skip user question (3).

Reply:
It will also ask for the credentials themselves, because on upgrades we do not have the admin password.

Reply:
No, you got it exactly right. My fault. For the internal engine admin password, we deliberately do not save it, because the user can change it later. Isn't it so also for keycloak?